src/main/java/com/google/codeu/servlets/MessageFeedServlet.java

```diff
@@ -18,26 +20,57 @@
 @WebServlet("/feed")
 public class MessageFeedServlet extends HttpServlet{
 
- private Datastore datastore;
+  private Datastore datastore;
 
- @Override
- public void init() {
-  datastore = new Datastore();
- }
+  @Override
+  public void init() {
+    datastore = new Datastore();
+  }
 
- /**
-  * Responds with a JSON representation of Message data for all users.
-  */
- @Override
- public void doGet(HttpServletRequest request, HttpServletResponse response)
-         throws IOException {
+  /**
+   * Responds with a JSON representation of Message data for all users.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
-  response.setContentType("application/json");
+    response.setContentType("application/json");
 
-  List<Message> messages = datastore.getAllMessages();
-  Gson gson = new Gson();
-  String json = gson.toJson(messages);
+    List<Message> messages = datastore.getAllMessages();
+    Gson gson = new Gson();
+    String json = gson.toJson(messages);
 
-  response.getOutputStream().println(json);
- }
+    response.setContentType("application/json; charset=UTF-8");
```

Comment:
Charset=UTF-8 and writer is used instead of the default outputstream so that characters in other languages can be recognised. However, the current implementation seems to be unable to interpret "&#39;" as "'". I am not sure how to resolve it. Please advise.

Reply:
Hmm I am not sure. So let's debug. If I gave it a string like `"'hello'"`, what's the JSON output?

Reply:
It seems that the bug only appears after translation. I.e, when I key in 你知道我在说什么吗, the translated message becomes "Do you know what \I&#39;m talking about". I am not very sure what is in the raw string.

Reply:
Ah I see. That's not really a problem with the translation but just that the strings are HTML encoded. They can be easily decoded in JS but you don't have to, as when they are viewed in the browser, the browser will automatically decode it. The reason why the strings are html encoded is to prevent security attacks: https://stackoverflow.com/questions/1014593/html-encode-what-how-does-it-prevent-scripting-security-problems-in-asp-net